src/vmspawn/vmspawn.c

```diff
@@ -473,7 +523,11 @@ static int run_virtual_machine(void) {
         if (r < 0)
                 return log_error_errno(r, "Failed to find QEMU binary: %m");
 
-        if (asprintf(&mem, "%.4fM", (double)arg_qemu_mem / (1024.0 * 1024.0)) < 0)
+        /* Apparently, qemu reads floating point numbers, hence let's be precise. We don't use floating %f
+         * floating point formatting here since it's locale dependent, and qemu doesn't want that. */
+        if (asprintf(&mem, "%zu.%05zuM",
```

Comment:
PRIu64 instead of zu?

Reply:
btw this ended up giving my problems in future so I just went with expressing it in megabytes as an integer: https://github.com/CodethinkLabs/systemd/blob/b2efd0d30180ba8d392d33c2f2abd7a15f86ac49/src/vmspawn/vmspawn.c#L901

This fixed an issue I had when this was introduced: https://github.com/CodethinkLabs/systemd/blob/b2efd0d30180ba8d392d33c2f2abd7a15f86ac49/src/vmspawn/vmspawn.c#L963

qemu seems to parse the value differently in the two places so expressing them as an integer seemed the simplest way to fix this.